driver-core/src/main/com/mongodb/operation/OperationHelper.java

```diff
@@ -274,6 +275,9 @@ static boolean canRetryWrite(final ServerDescription serverDescription, final Co
         } else if (connectionDescription.getServerType().equals(ServerType.STANDALONE)) {
             LOGGER.debug("retryWrites set to true but the server is a standalone server.");
             return false;
+        } else if (!sessionContext.hasSession()) {
+            LOGGER.debug("retryWrites set to true but there is no implicit session.");
```

Comment:
The only reason I'm aware of that this can happen is if there are multiple users authenticated, but there's no way to check that directly here, so making the log message rather generic.

Reply:
Could we indicate that might be the cause? So users have an idea of how to fix?

Reply:
Done